### PR TITLE
test: benchmark sdk profiling components

### PR DESCRIPTION
**Pull request by armcknight**
## :scroll: Description

#skip-changelog

See additions to the [develop-docs README](https://github.com/getsentry/sentry-cocoa/pull/1834/files#diff-5c546014845a9fa5c3f1d7c177d42608d616f3206059c5efac54404ffcc390e8) for explanation.

## :bulb: Motivation and Context

We want to have instrumentation in place to measure the CPU overhead of profiling components in the SDK, both as a guardrail for PRs, to let us know if we need to optimize now and by how much, and for us to observe the trend over time.

## :green_heart: How did you test it?

Run the new benchmark tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: develop-docs/README.md
Comment (philipphofmann) [inline develop-docs/README.md:90]: I guess we miss how many times.

File: .github/workflows/performance-benchmarks.yml
Comment (philipphofmann) [inline .github/workflows/performance-benchmarks.yml:18]: Didn't know that there is v3 here already. We can update our other actions https://github.com/getsentry/sentry-cocoa/pull/1845

File: .github/workflows/performance-benchmarks.yml
Comment (philipphofmann) [inline .github/workflows/performance-benchmarks.yml:21]: Same for this one https://github.com/getsentry/sentry-cocoa/pull/1846

File: Makefile
Comment (philipphofmann) [inline Makefile:101]: Maybe you want to store the raw test log into a file and upload it as an artifact if the tests fail. This is helpful when investigations flaky tests
https://github.com/getsentry/sentry-cocoa/blob/2d7cc7b287795b8d567643a870a4ef1067825dfe/scripts/xcode-test.sh#L68
https://github.com/getsentry/sentry-cocoa/blob/2d7cc7b287795b8d567643a870a4ef1067825dfe/.github/workflows/test.yml#L110-L116

File: Samples/TrendingMovies/TrendingMovies/ImageProcessing/ImageEffects.swift
Comment (philipphofmann) [inline Samples/TrendingMovies/TrendingMovies/ImageProcessing/ImageEffects.swift:15]: `l`: Don't we want to delete this code?

File: Samples/TrendingMovies/TrendingMovies/Utilities/Tracer.swift
Comment (philipphofmann) [inline Samples/TrendingMovies/TrendingMovies/Utilities/Tracer.swift:37]: `m`: Don't we want to have this enabled by default for playing around with the sample app? We could invert this flag and pass an argument when we want disable it.

File: Samples/TrendingMovies/TrendingMovies/SentryBenchmarking.mm
Comment (philipphofmann) [inline Samples/TrendingMovies/TrendingMovies/SentryBenchmarking.mm:16]: If this is ported from SentryThreadHandle.hpp I guess this was already reviewed, so I'm not going to review it again.

**Review by philipphofmann — COMMENTED** [review]
There is some commented code in the PR. Did you miss something @armcknight or forgot to delete it? Apart from that almost LGTM.

File: Samples/TrendingMovies/TrendingMovies/Utilities/Tracer.swift
Comment (armcknight) [reply Samples/TrendingMovies/TrendingMovies/Utilities/Tracer.swift:37]: Good call.

File: Samples/TrendingMovies/TrendingMovies/ImageProcessing/ImageEffects.swift
Comment (armcknight) [reply Samples/TrendingMovies/TrendingMovies/ImageProcessing/ImageEffects.swift:15]: Maybe I'll also make these dependent on the testing launch arg.

File: Makefile
Comment (armcknight) [reply Makefile:101]: Thanks. My current issues aren't for a lack of logging 😅

**Review by philipphofmann — COMMENTED** [review]
One major comment, expect from that LGTM.

File: .github/workflows/ui-tests.yml
Comment (armcknight) [reply .github/workflows/ui-tests.yml:82]: I removed this for debugging purposes as saucelabs runs were taking about an hour, I'll revert the change before merging.

File: Samples/iOS-Swift/iOS-SwiftUITests/SDKPerformanceBenchmarkTests.swift
Comment (armcknight) [reply Samples/iOS-Swift/iOS-SwiftUITests/SDKPerformanceBenchmarkTests.swift:6]: yeah this is pointless since there's only one test case here anyways... no reason other than that it is from the default UI test template :)

File: Samples/iOS-Swift/iOS-SwiftUITests/SDKPerformanceBenchmarkTests.swift
Comment (philipphofmann) [inline Samples/iOS-Swift/iOS-SwiftUITests/SDKPerformanceBenchmarkTests.swift:5]: `l`: Why only on iOS 15?

**Review by philipphofmann — APPROVED** [review]
LGTM 🚀

File: Samples/iOS-Swift/iOS-SwiftUITests/SDKPerformanceBenchmarkTests.swift
Comment (armcknight) [reply Samples/iOS-Swift/iOS-SwiftUITests/SDKPerformanceBenchmarkTests.swift:5]: After discussing the issue with iOS 10, we decided we can start by only running the benchmark on latest iOS. If we want to look for variations later due to API differences we can make changes here.

**Comment by armcknight** [comment]
The test currently times out saucelabs as it only allows for 15 min per test method by default. Once I can resolve that issue I’ll squash and merge the PR @philipphofmann

**Comment by armcknight** [comment]
@philipphofmann please wait to merge this PR until after merging https://github.com/getsentry/sentry-cocoa/pull/1862; that has some test fixes needed for this PR.

**Comment by armcknight** [comment]
The only failing check is for a [CodeQL security check](https://github.com/getsentry/sentry-cocoa/security/code-scanning/1) that's not relevant to these changes, will take up in subsequent work.

**Pull request merged**
